cli/cmd/agent_install_ec2ssh.go

```diff
@@ -107,44 +109,60 @@ func installAWSSSH(_ *cobra.Command, _ []string) error {
 		return err
 	}
 
+	wg := new(sync.WaitGroup)
+	cl := limiter.NewConcurrencyLimiter(agentCmdState.InstallMaxParallelism)
 	for _, runner := range runners {
-		cli.Log.Debugw("runner info: ",
-			"user", runner.Runner.User,
-			"region", runner.Region,
-			"az", runner.AvailabilityZone,
-			"instance ID", runner.InstanceID,
-			"hostname", runner.Runner.Hostname,
-		)
-
-		err := runner.Runner.UseIdentityFile(agentCmdState.InstallIdentityFile)
-		if err != nil {
-			return errors.Wrap(err, "unable to use provided identity file")
-		}
+		wg.Add(1)
+
+		// In order to use `cl.Execute()`, the input func() must not take any arguments.
+		// Copy the runner info to dedicated variable in the goroutine to prevent race overwrite
+		runnerCopyWg := new(sync.WaitGroup)
+		runnerCopyWg.Add(1)
+
+		_, err := cl.Execute(func() {
+			threadRunner := *runner
+			runnerCopyWg.Done()
+			cli.Log.Debugw("threadRunner info: ",
+				"user", threadRunner.Runner.User,
+				"region", threadRunner.Region,
+				"az", threadRunner.AvailabilityZone,
+				"instance ID", threadRunner.InstanceID,
+				"hostname", threadRunner.Runner.Hostname,
+			)
+
+			err := threadRunner.Runner.UseIdentityFile(agentCmdState.InstallIdentityFile)
+			if err != nil {
+				cli.Log.Warnw("unable to use provided identity file", "err", err, "threadRunner", threadRunner.InstanceID)
```

Comment:
the standard for fields in the CLI log entries is `snake_case` rather than `CamelCase`
```suggestion
				cli.Log.Warnw("unable to use provided identity file", "err", err, "thread_runner", threadRunner.InstanceID)
```

Reply:
Fixed with eb423daaf99b4291448e156ea4ffe8d3e0e0a3af